woocommerce/wc-admin
# Register core product taxonomies

Fixes #5795

* Adds methods to register and detect taxonomy screens.
* Registers core taxonomies and custom product attributes

~~Note that if you create a custom product atribute and visit that page (e.g., `wp-admin/edit-tags.php?taxonomy=pa_test-attr&post_type=product`) that the menu item is not correctly matched.  Going to leave this for a follow-up since decisions made in https://github.com/woocommerce/woocommerce-admin/pull/5798 may affect this.~~ - Fixed by regex approach.

### Screenshots
<img width="495" alt="Screen Shot 2020-12-03 at 5 20 37 PM" src="https://user-images.githubusercontent.com/10561050/101095694-51437100-358c-11eb-8689-706a97fe5fa8.png">


### Detailed test instructions:

1. Enable the nav feature.
1. Navigate to the "Products" category.
1. Make sure the expected items and appear.
1. Make sure navigating to each of the taxonomies and individual term pages works as expected (with the exception of the custom PA mentioned above).
